src/imgui/ImGuiMachine.cc

```diff
@@ -46,8 +46,69 @@ ImGuiMachine::ImGuiMachine(ImGuiManager& manager_)
 	, setupFileList("setup", Reactor::SETUP_EXTENSION, Reactor::SETUP_DIR)
 	, confirmDialog("Confirm##setup")
 {
-	setupFileList.selectAction = [&](const FileListWidget::Entry& entry) {
-		manager.executeDelayed(makeTclList("setup", entry.displayName));
+
+	setupFileList.drawAction = [&] {
+		im::Table("table", 2, ImGuiTableFlags_BordersInnerV, [&]{
+			if (ImGui::TableNextColumn()) {
+				setupFileList.drawTable();
+			}
+			if (ImGui::TableNextColumn()) {
+				if (previewSetup.motherBoard) {
+					showSetupOverview(*previewSetup.motherBoard);
+					ImGui::Separator();
+					if (ImGui::Button("Run!")) {
+						manager.executeDelayed([&previewSetup = previewSetup, &manager = manager] {
+							manager.getReactor().switchMachineFromSetup(previewSetup.fullName);
```

Comment:
Did you try what happens when this (delayed) action fails?
I assume you must still handle the exception?

Reply:
Good point, I forgot about that. Fixed.